Indicators/ChaikinMoneyFlow.cs

```diff
@@ -0,0 +1,107 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Collections.Generic;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The Chaikin Money Flow Index (CMF) is a volume-weighted average of accumulation and distribution over
+    /// a specified period.
+    ///
+    /// CMF = n-day Sum of [(((C - L) - (H - C)) / (H - L)) x Vol] / n-day Sum of Vol
+    ///
+    /// Where:
+    /// n = number of periods, typically 21
+    /// H = high
+    /// L = low
+    /// C = close
+    /// Vol = volume
+    /// 
+    /// https://www.fidelity.com/learning-center/trading-investing/technical-analysis/technical-indicator-guide/cmf
+    /// </summary>
+    public class ChaikinMoneyFlow : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// Holds the point-wise flow and volume terms. 
+        /// </summary>
+        private readonly RollingWindow<List<decimal>> _rollingFlow;
+
+        private Sum _flowRatioSum;
+        private Sum _volumeSum;
```

Comment:
These private fields could be `readonly`, since they are defined only once in the constructor.